File: PAMO/app/src/main/java/pl/pamo/ui/chart/ChartFragment.java
package pl.pamo.ui.chart;

import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import com.github.mikephil.charting.charts.LineChart;
import com.github.mikephil.charting.components.XAxis;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.data.LineData;
import com.github.mikephil.charting.data.LineDataSet;
import com.github.mikephil.charting.formatter.ValueFormatter;

import org.jetbrains.annotations.NotNull;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Locale;

import pl.pamo.R;

public class ChartFragment extends Fragment {


    private LineChart chart;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {


        View rootView = inflater.inflate(R.layout.fragment_charts, container, false);

        chart = rootView.findViewById(R.id.chart);

        createRender();

        return rootView;
    }

    private void createRender() {


        LineDataSet set = new LineDataSet(getCovidDataSet(), "COVID-19 cases");

        set.setDrawValues(false);
        set.setLineWidth(3);
        set.setColor(Color.GREEN);
        set.setCircleColor(Color.GREEN);

        LineData lineData = new LineData(set);
        chart.setData(lineData);
        chart.invalidate();

        XAxis xAxis = chart.getXAxis();
        xAxis.setValueFormatter(new ValueFormatter() {
            private final SimpleDateFormat mFormat = new SimpleDateFormat("dd MMM yyyy", Locale.ENGLISH);

            @Override
            public String getFormattedValue(float value) {
                long millis = (long) value;
                return mFormat.format(new Date(millis));
            }
        });
    }

    @NotNull
    private List<Entry> getCovidDataSet() {
        List<Entry> entryList = new ArrayList<>();


        entryList.add(new Entry(convertDateToMillis("30 December 2019"), 1));
        entryList.add(new Entry(convertDateToMillis("6 January  2020"), 45));
        entryList.add(new Entry(convertDateToMillis("13 January 2020"), 137));
        entryList.add(new Entry(convertDateToMillis("20 January 2020"), 2036));
        entryList.add(new Entry(convertDateToMillis("27 January 2020"), 14582));
        entryList.add(new Entry(convertDateToMillis("3 February 2020"), 37588));
        entryList.add(new Entry(convertDateToMillis("10 February 2020"), 69307));
        entryList.add(new Entry(convertDateToMillis("17 February 2020"), 78890));
        entryList.add(new Entry(convertDateToMillis("27 February 2020"), 87146));
        entryList.add(new Entry(convertDateToMillis("2 March 2020"), 107368));
        entryList.add(new Entry(convertDateToMillis("9 March 2020"), 160971));
        entryList.add(new Entry(convertDateToMillis("16 March 2020"), 314597));
        entryList.add(new Entry(convertDateToMillis("23 March 2020"), 675472));
        entryList.add(new Entry(convertDateToMillis("30 March 2020"), 1172724));
        entryList.add(new Entry(convertDateToMillis("6 April 2020"), 1727065));
        entryList.add(new Entry(convertDateToMillis("13 April 2020"), 2260197));
        entryList.add(new Entry(convertDateToMillis("20 April 2020"), 2813992));
        entryList.add(new Entry(convertDateToMillis("27 April 2020"), 3361867));
        entryList.add(new Entry(convertDateToMillis("4 May 2020"), 3944859));
        entryList.add(new Entry(convertDateToMillis("11 May 2020"), 4542228));
        entryList.add(new Entry(convertDateToMillis("18 May 2020"), 5221880));
        entryList.add(new Entry(convertDateToMillis("25 May 2020"), 5941401));
        entryList.add(new Entry(convertDateToMillis("1 June 2020"), 6784151));
        entryList.add(new Entry(convertDateToMillis("8 June 2020"), 7672904));
        entryList.add(new Entry(convertDateToMillis("15 June 2020"), 8692127));
        entryList.add(new Entry(convertDateToMillis("22 June 2020"), 9831317));
        entryList.add(new Entry(convertDateToMillis("29 June 2020"), 11136747));
        entryList.add(new Entry(convertDateToMillis("6 July 2020"), 12568287));
        entryList.add(new Entry(convertDateToMillis("13 July 2020"), 14128559));
        entryList.add(new Entry(convertDateToMillis("20 July 2020"), 14128559));
        entryList.add(new Entry(convertDateToMillis("27 July 2020"), 17726008));
        entryList.add(new Entry(convertDateToMillis("3 August 2020"), 19563096));
        entryList.add(new Entry(convertDateToMillis("10 August 2020"), 21563096));
        entryList.add(new Entry(convertDateToMillis("17 August 2020"), 23265767));
        entryList.add(new Entry(convertDateToMillis("24 August 2020"), 25120592));
        entryList.add(new Entry(convertDateToMillis("31 August 2020"), 27086891));
        entryList.add(new Entry(convertDateToMillis("7 September 2020"), 29025081));
        entryList.add(new Entry(convertDateToMillis("14 September 2020"), 31117228));
        entryList.add(new Entry(convertDateToMillis("21 September 2020"), 33222965));
        entryList.add(new Entry(convertDateToMillis("28 September 2020"), 35363453));
        entryList.add(new Entry(convertDateToMillis("5 October 2020"), 37727659));
        entryList.add(new Entry(convertDateToMillis("12 October 2020"), 40286863));
        entryList.add(new Entry(convertDateToMillis("19 October 2020"), 43307038));
        entryList.add(new Entry(convertDateToMillis("26 October 2020"), 46766320));
        entryList.add(new Entry(convertDateToMillis("2 November 2020"), 50558407));
        entryList.add(new Entry(convertDateToMillis("9 November 2020"), 54636968));
        entryList.add(new Entry(convertDateToMillis("16 November 2020"), 58770226));
        entryList.add(new Entry(convertDateToMillis("23 November 2020"), 62828095));
        entryList.add(new Entry(convertDateToMillis("30 November 2020"), 67009710));
        entryList.add(new Entry(convertDateToMillis("7 December 2020"), 71389145));
        entryList.add(new Entry(convertDateToMillis("14 December 2020"), 76081004));
        entryList.add(new Entry(convertDateToMillis("21 December 2020"), 80203399));
        entryList.add(new Entry(convertDateToMillis("28 December 2020"), 84372702));
        entryList.add(new Entry(convertDateToMillis("4 January 2021"), 89417030));
        entryList.add(new Entry(convertDateToMillis("11 January 2021"), 94249569));
        entryList.add(new Entry(convertDateToMillis("18 January 2021"), 98497983));
        entryList.add(new Entry(convertDateToMillis("25 January 2021"), 102282845));
        entryList.add(new Entry(convertDateToMillis("1 February 2021"), 105506243));
        entryList.add(new Entry(convertDateToMillis("8 February 2021"), 108266521));
        entryList.add(new Entry(convertDateToMillis("15 February 2021"), 110757376));
        entryList.add(new Entry(convertDateToMillis("22 February 2021"), 113442688));
        entryList.add(new Entry(convertDateToMillis("1 March 2021"), 116198244));
        entryList.add(new Entry(convertDateToMillis("8 March 2021"), 119245795));
        entryList.add(new Entry(convertDateToMillis("15 March 2021"), 122870978));
        entryList.add(new Entry(convertDateToMillis("22 March 2021"), 126405434));
        entryList.add(new Entry(convertDateToMillis("29 March 2021"), 130508795));
        entryList.add(new Entry(convertDateToMillis("5 April 2021"), 135097828));
        entryList.add(new Entry(convertDateToMillis("12 April 2021"), 140373678));
        entryList.add(new Entry(convertDateToMillis("19 April 2021"), 146105297));
        entryList.add(new Entry(convertDateToMillis("26 April 2021"), 151845101));
        entryList.add(new Entry(convertDateToMillis("3 May 2021"), 157269221));
        entryList.add(new Entry(convertDateToMillis("10 May 2021"), 160076944));
        return entryList;
    }

    long convertDateToMillis(String date) {
        SimpleDateFormat f = new SimpleDateFormat("dd MMM yyyy");
        Date d = null;

        try {
            d = f.parse(date);
        } catch (ParseException e) {
            e.printStackTrace();
        }

        return d.getTime();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }
}

File: PAMO/app/src/main/java/ui/chart/ChartFragment.kt
package ui.chart

import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.github.mikephil.charting.charts.LineChart
import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.data.LineData
import com.github.mikephil.charting.data.LineDataSet
import com.github.mikephil.charting.formatter.ValueFormatter
import pl.pamo.R
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

class ChartFragment : Fragment() {
    private var chart: LineChart? = null
    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_charts, container, false)
        chart = rootView.findViewById(R.id.chart)
        createRender()
        return rootView
    }

    private fun createRender() {
        val set = LineDataSet(covidDataSet, "COVID-19 cases")
        set.setDrawValues(false)
        set.lineWidth = 3f
        set.color = Color.GREEN
        set.setCircleColor(Color.GREEN)
        val lineData = LineData(set)
        chart!!.data = lineData
        chart!!.invalidate()
        val xAxis = chart!!.xAxis
        xAxis.valueFormatter = object : ValueFormatter() {
            private val mFormat = SimpleDateFormat("dd MMM yyyy", Locale.ENGLISH)
            override fun getFormattedValue(value: Float): String {
                val millis = value.toLong()
                return mFormat.format(Date(millis))
            }
        }
    }

    private val covidDataSet: List<Entry>
        private get() {
            val entryList: MutableList<Entry> = ArrayList()
            entryList.add(Entry(convertDateToMillis("30 December 2019").toFloat(), 1F))
            entryList.add(Entry(convertDateToMillis("6 January  2020").toFloat(), 45F))
            entryList.add(Entry(convertDateToMillis("13 January 2020").toFloat(), 137F))
            entryList.add(Entry(convertDateToMillis("20 January 2020").toFloat(), 2036F))
            entryList.add(Entry(convertDateToMillis("27 January 2020").toFloat(), 14582F))
            entryList.add(Entry(convertDateToMillis("3 February 2020").toFloat(), 37588F))
            entryList.add(Entry(convertDateToMillis("10 February 2020").toFloat(), 69307F))
            entryList.add(Entry(convertDateToMillis("17 February 2020").toFloat(), 78890F))
            entryList.add(Entry(convertDateToMillis("27 February 2020").toFloat(), 87146F))
            entryList.add(Entry(convertDateToMillis("2 March 2020").toFloat(), 107368F))
            entryList.add(Entry(convertDateToMillis("9 March 2020").toFloat(), 160971F))
            entryList.add(Entry(convertDateToMillis("16 March 2020").toFloat(), 314597F))
            entryList.add(Entry(convertDateToMillis("23 March 2020").toFloat(), 675472F))
            entryList.add(Entry(convertDateToMillis("30 March 2020").toFloat(), 1172724F))
            entryList.add(Entry(convertDateToMillis("6 April 2020").toFloat(), 1727065F))
            entryList.add(Entry(convertDateToMillis("13 April 2020").toFloat(), 2260197F))
            entryList.add(Entry(convertDateToMillis("20 April 2020").toFloat(), 2813992F))
            entryList.add(Entry(convertDateToMillis("27 April 2020").toFloat(), 3361867F))
            entryList.add(Entry(convertDateToMillis("4 May 2020").toFloat(), 3944859F))
            entryList.add(Entry(convertDateToMillis("11 May 2020").toFloat(), 4542228F))
            entryList.add(Entry(convertDateToMillis("18 May 2020").toFloat(), 5221880F))
            entryList.add(Entry(convertDateToMillis("25 May 2020").toFloat(), 5941401F))
            entryList.add(Entry(convertDateToMillis("1 June 2020").toFloat(), 6784151F))
            entryList.add(Entry(convertDateToMillis("8 June 2020").toFloat(), 7672904F))
            entryList.add(Entry(convertDateToMillis("15 June 2020").toFloat(), 8692127F))
            entryList.add(Entry(convertDateToMillis("22 June 2020").toFloat(), 9831317F))
            entryList.add(Entry(convertDateToMillis("29 June 2020").toFloat(), 11136747F))
            entryList.add(Entry(convertDateToMillis("6 July 2020").toFloat(), 12568287F))
            entryList.add(Entry(convertDateToMillis("13 July 2020").toFloat(), 14128559F))
            entryList.add(Entry(convertDateToMillis("20 July 2020").toFloat(), 14128559F))
            entryList.add(Entry(convertDateToMillis("27 July 2020").toFloat(), 17726008F))
            entryList.add(Entry(convertDateToMillis("3 August 2020").toFloat(), 19563096F))
            entryList.add(Entry(convertDateToMillis("10 August 2020").toFloat(), 21563096F))
            entryList.add(Entry(convertDateToMillis("17 August 2020").toFloat(), 23265767F))
            entryList.add(Entry(convertDateToMillis("24 August 2020").toFloat(), 25120592F))
            entryList.add(Entry(convertDateToMillis("31 August 2020").toFloat(), 27086891F))
            entryList.add(Entry(convertDateToMillis("7 September 2020").toFloat(), 29025081F))
            entryList.add(Entry(convertDateToMillis("14 September 2020").toFloat(), 31117228F))
            entryList.add(Entry(convertDateToMillis("21 September 2020").toFloat(), 33222965F))
            entryList.add(Entry(convertDateToMillis("28 September 2020").toFloat(), 35363453F))
            entryList.add(Entry(convertDateToMillis("5 October 2020").toFloat(), 37727659F))
            entryList.add(Entry(convertDateToMillis("12 October 2020").toFloat(), 40286863F))
            entryList.add(Entry(convertDateToMillis("19 October 2020").toFloat(), 43307038F))
            entryList.add(Entry(convertDateToMillis("26 October 2020").toFloat(), 46766320F))
            entryList.add(Entry(convertDateToMillis("2 November 2020").toFloat(), 50558407F))
            entryList.add(Entry(convertDateToMillis("9 November 2020").toFloat(), 54636968F))
            entryList.add(Entry(convertDateToMillis("16 November 2020").toFloat(), 58770226F))
            entryList.add(Entry(convertDateToMillis("23 November 2020").toFloat(), 62828095F))
            entryList.add(Entry(convertDateToMillis("30 November 2020").toFloat(), 67009710F))
            entryList.add(Entry(convertDateToMillis("7 December 2020").toFloat(), 71389145F))
            entryList.add(Entry(convertDateToMillis("14 December 2020").toFloat(), 76081004F))
            entryList.add(Entry(convertDateToMillis("21 December 2020").toFloat(), 80203399F))
            entryList.add(Entry(convertDateToMillis("28 December 2020").toFloat(), 84372702F))
            entryList.add(Entry(convertDateToMillis("4 January 2021").toFloat(), 89417030F))
            entryList.add(Entry(convertDateToMillis("11 January 2021").toFloat(), 94249569F))
            entryList.add(Entry(convertDateToMillis("18 January 2021").toFloat(), 98497983F))
            entryList.add(Entry(convertDateToMillis("25 January 2021").toFloat(), 102282845F))
            entryList.add(Entry(convertDateToMillis("1 February 2021").toFloat(), 105506243F))
            entryList.add(Entry(convertDateToMillis("8 February 2021").toFloat(), 108266521F))
            entryList.add(Entry(convertDateToMillis("15 February 2021").toFloat(), 110757376F))
            entryList.add(Entry(convertDateToMillis("22 February 2021").toFloat(), 113442688F))
            entryList.add(Entry(convertDateToMillis("1 March 2021").toFloat(), 116198244F))
            entryList.add(Entry(convertDateToMillis("8 March 2021").toFloat(), 119245795F))
            entryList.add(Entry(convertDateToMillis("15 March 2021").toFloat(), 122870978F))
            entryList.add(Entry(convertDateToMillis("22 March 2021").toFloat(), 126405434F))
            entryList.add(Entry(convertDateToMillis("29 March 2021").toFloat(), 130508795F))
            entryList.add(Entry(convertDateToMillis("5 April 2021").toFloat(), 135097828F))
            entryList.add(Entry(convertDateToMillis("12 April 2021").toFloat(), 140373678F))
            entryList.add(Entry(convertDateToMillis("19 April 2021").toFloat(), 146105297F))
            entryList.add(Entry(convertDateToMillis("26 April 2021").toFloat(), 151845101F))
            entryList.add(Entry(convertDateToMillis("3 May 2021").toFloat(), 157269221F))
            entryList.add(Entry(convertDateToMillis("10 May 2021").toFloat(), 160076944F))
            return entryList
        }

    fun convertDateToMillis(date: String?): Long {
        val f = SimpleDateFormat("dd MMM yyyy")
        var d: Date? = null
        try {
            d = f.parse(date)
        } catch (e: ParseException) {
            e.printStackTrace()
        }
        return d!!.time
    }

    override fun onDestroyView() {
        super.onDestroyView()
    }
}